opentuner/search/technique.py

```diff
@@ -204,36 +228,67 @@ def call_main_generator(self):
                                     self.pending_tests)
         if self.pending_tests:
           self.rounds_since_novel_request = 0
-          yield None # wait
+          yield False # wait
 
 #list of all techniques
 the_registry = list()
 
+#list of technique generators
+the_generator_registry = list()
 
 def register(t):
   the_registry.append(t)
 
+def register_generator(cls, generator_weight=1.0, *args, **kwargs):
+  """
+  register a technique generator - a tuple of (technique class, args, kwargs)
```

Comment:
Updated this in the last commit. Let me know if there's anything else, otherwise I can squash everything then try to split it up into separate commits for each thing (suggested by Jeffrey), e.g. generator stuff, the greedy technique, resetting sequential search techniques, etc